snapshots/devmapper/snapshotter.go

```diff
@@ -398,26 +410,86 @@ func (s *Snapshotter) createSnapshot(ctx context.Context, kind snapshots.Kind, k
 	return mounts, nil
 }
 
-// mkfs creates ext4 filesystem on the given devmapper device
-func mkfs(ctx context.Context, path string) error {
-	args := []string{
-		"-E",
-		// We don't want any zeroing in advance when running mkfs on thin devices (see "man mkfs.ext4")
-		"nodiscard,lazy_itable_init=0,lazy_journal_init=0",
-		path,
+// mkfs creates filesystem on the given devmapper device based on type
+// specified in config.
+func mkfs(ctx context.Context, fs fsType, path string) error {
+	mkfsCommand := ""
+	var args []string
+
+	switch fs {
+	case fsTypeExt4:
+		mkfsCommand = "mkfs.ext4"
+		args = []string{
+			"-E",
+			// We don't want any zeroing in advance when running mkfs on thin devices (see "man mkfs.ext4")
+			"nodiscard,lazy_itable_init=0,lazy_journal_init=0",
+			path,
+		}
+	case fsTypeXfs:
+		mkfsCommand = "mkfs.xfs"
+		args = []string{
+			path,
+		}
+	default:
+		return errors.New("file system not supported")
 	}
 
-	log.G(ctx).Debugf("mkfs.ext4 %s", strings.Join(args, " "))
-	b, err := exec.Command("mkfs.ext4", args...).CombinedOutput()
+	log.G(ctx).Debugf("%s %s", mkfsCommand, strings.Join(args, " "))
+	b, err := exec.Command(mkfsCommand, args...).CombinedOutput()
 	out := string(b)
 	if err != nil {
-		return errors.Wrapf(err, "mkfs.ext4 couldn't initialize %q: %s", path, out)
+		return errors.Wrapf(err, "%s couldn't initialize %q: %s", mkfsCommand, path, out)
 	}
 
 	log.G(ctx).Debugf("mkfs:\n%s", out)
 	return nil
 }
 
+func (s *Snapshotter) getFsType(device string) string {
```

Comment:
I'm not sure this is really necessary.
We can add a label to `snapshots.Info.Labels` that can indicate file system type we use (when creating a new snapshot).
If there is no such label, we just default to `ext4` (or another way - do a small migration step at snapshotter init, iterate over all snapshots without label and append it).

Reply:
I was thinking that, reading from the device would be much simpler than maintaining/reading additional metadata? Reading from the device didn't seem to be time consuming for an earlier experiment. The buildMount function with getFsType took around 95886 nanoseconds. Do you have any specific concerns? Having labels is definitely the other alternative.

Reply:
`Mounts` can be called many times (for example, have a look calls in `oci/spec_opts.go`, it's quite possible to have more than one call within same OCI spec configuration), so we'll have to read those volumes every time.
Keeping this info in metadata is preferable IMO because that's why we maintain metadata database in the first place.
Also the approach with label seems like easier to maintain.

Reply:
96ms seems like a decent amount of overhead if we're reading somewhat frequently

Reply:
updated the PR. Using lables for file system info.